src/classes/RD2_ElevateIntegrationService.cls

```diff
@@ -145,11 +148,9 @@ public inherited sharing class RD2_ElevateIntegrationService {
 
         if (!isChanged) {
             String oldCommitmentId = oldRd.CommitmentId__c;
-            Boolean isTempIdOverwrite = String.isNotBlank(oldCommitmentId)
-                && oldCommitmentId.startsWith(TEMP_COMMITMENT_ID_PREFIX);
 
-            isChanged = rd.CommitmentId__c != oldCommitmentId
-                && !isTempIdOverwrite;
+            isChanged = String.isNotBlank(oldCommitmentId)
```

Comment:
Temp commitment Id is not relevant any longer since commitment will be created before the RD. (W-8554656)